src/algorithm/naive.rs

```diff
@@ -82,22 +99,91 @@ where
 
             let mut thread_rng: RNG = SeedableRng::from_rng(rng.clone())?;
 
-            for _ in 0..(*ledger_view.get_random_walks_num()) {
+            for _ in 0..r_value {
                 let mut walk = RandomWalk::new(i.clone());
-                let mut current_node = i;
-                // TODO distinguish account/project
+                let mut current_node_id = i;
+
+                let current_node = network
+                    .get_node(current_node_id)
+                    .expect("Couldn't access not during random walk.");
+
+                let damping_factor = match &current_node.node_type() {
+                    types::NodeType::User => acc_epsilon,
+                    types::NodeType::Project => prj_epsilon,
+                };
+
                 // TODO Should there be a safeguard so this doesn't run forever?
-                while thread_rng.gen::<f64>() < ledger_view.get_damping_factors().project {
-                    let neighbors = network.edges_directed(&current_node, Direction::Outgoing);
-                    match neighbors.choose_weighted(&mut thread_rng, |item| {
-                        network
-                            .get_edge(&item.id)
-                            .and_then(|m| Some(m.weight()))
-                            .unwrap()
-                    }) {
-                        Ok(next_edge) => {
-                            walk.add_next(next_edge.to.clone());
-                            current_node = next_edge.to;
+                while thread_rng.gen::<f64>() < damping_factor {
```

Comment:
The `damping_factor` depends on the current node, so has to be changed according to on which node type the random walker is.

Reply:
Ah, drat, you are right!

Reply:
Self-answering: We need to push the `current_node` and `damping_factor` inside the while loop, otherwise we would lock these values once for all.